test/e2e/redundancy/gateway_failover.go

```diff
@@ -87,7 +87,8 @@ func testGatewayPodRestartScenario(f *subFramework.Framework) {
 
 	framework.By(fmt.Sprintf("Ensuring that the gateway reports as active on %q", primaryClusterName))
 
-	activeGateway := f.AwaitGatewayFullyConnected(primaryCluster, resource.EnsureValidName(gatewayNodes[0].Name))
```

Comment:
Maybe leave a comment here to explain why hostname is retrieved from endpoint instead of gatewayNodes[0].Name